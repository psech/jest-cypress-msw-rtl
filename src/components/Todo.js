import React from 'react';

export default function Todo({ todo, onChange }) {
  return (
    <tr>
      <td>{todo.id}</td>
      <td>{todo.title}</td>
      <td>
        <input
          type="checkbox"
          defaultChecked={todo.completed}
          onChange={() => onChange(todo.id)}
        />
      </td>
    </tr>
  );
}
