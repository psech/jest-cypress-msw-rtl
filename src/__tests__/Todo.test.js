import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Todo from '../components/Todo';

describe('Todo', () => {
  test('Should match snapshot', () => {
    const todo = {
      userId: 10,
      id: 10,
      title: 'some title',
      completed: false,
    };

    const handleChange = jest.fn();

    const { container } = render(
      <Todo key={todo.id} todo={todo} onChange={handleChange} />
    );

    const row = container.querySelector('tr');
    expect(row).toMatchSnapshot();
  });

  test('Handle select checkbox', () => {
    const todo = {
      userId: 10,
      id: 10,
      title: 'some title',
      completed: false,
    };

    const handleChange = jest.fn();

    const { container } = render(
      <Todo key={todo.id} todo={todo} onChange={handleChange} />
    );

    userEvent.click(container.querySelector('input'));

    expect(handleChange).toHaveBeenCalledTimes(1);
  });
});
