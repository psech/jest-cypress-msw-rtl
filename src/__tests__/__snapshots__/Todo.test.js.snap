// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Should match snapshot 1`] = `
<tr>
  <td>
    10
  </td>
  <td>
    some title
  </td>
  <td>
    <input
      type="checkbox"
    />
  </td>
</tr>
`;
